test/metabase/test/util.clj

```diff
@@ -264,6 +264,11 @@
   [obj]
   (json/parse-string (json/generate-string obj) keyword))
 
+(defn int->kw
+  "Convert an integer to a keyword. I.e : 3 -> :3"
+  [x]
+  (keyword (str x)))
```

Comment:
Hmm...having to create keywords out of ints is a code smell, but it seems like this is a side effect of using `keyword` in the test HTTP client to [parse all JSON keys](https://github.com/metabase/metabase/blob/60bfbc61b94df6915ea6fd73f83046d08cf5fa7c/test/metabase/http_client.clj#L85). It hasn't been an issue before because we just don't have API-level tests for the other graph fetching endpoints.

Maybe you could try changing that HTTP client parsing code to parse ints properly in the response, rather than indiscriminately making them keywords? If that works and doesn't break a ton of other tests, it's a more future-proof solution.

Reply:
Nice catch, but I lets do it in a another PR, I tried to parse it but we have 13 failing tests.
[Change](https://github.com/metabase/metabase/pull/21124/commits/da4105947fad41ae43293b9a33ed94dcb3df76b2)
[CI](https://app.circleci.com/pipelines/github/metabase/metabase/31543/workflows/2ec0220f-9552-437e-90bf-d35577edd4c9/jobs/1599343)